src/AISettings.cpp

```diff
@@ -0,0 +1,169 @@
+//
+// Created by rolf on 07-03-21.
+//
+
+#include "AISettings.h"
+#include <roboteam_utils/Print.h>
+
+namespace rtt {
+AISettings::AISettings(int initial_id) :
+    id{initial_id},
+    is_paused(true),
+    listenToReferee(false){
+  //these settings are for testing and convenience, basically
+  if (initial_id == 1) {
+    is_yellow = false;
+    is_left = false;
+  } else {
+    is_yellow = true;
+    is_left = true;
+  }
+}
+
+int AISettings::getId() const {
+  return id;
+}
+
+bool AISettings::isYellow() const {
+  return is_yellow;
+}
+void AISettings::setYellow(bool isYellow) {
+  is_yellow = isYellow;
+
+}
+bool AISettings::isLeft() const {
+  return is_left;
+}
+void AISettings::setLeft(bool left) {
+  is_left = left;
+}
+
+
+void AISettings::setPause(bool paused) {
+  is_paused = paused; //TODO: when the AI is just paused, send a halt command to all robots
+}
+bool AISettings::isPaused() const {
+  return is_paused;
+}
+bool AISettings::getListenToReferee() const {
+  return listenToReferee;
+}
+void AISettings::setListenToReferee(bool listen) {
+  listenToReferee = listen;
+}
+
+std::string AISettings::name() const {
+  return "ai"+std::to_string(id);
+}
+proto::Handshake AISettings::getButtonDeclarations() const{
```

Comment:
Wouldn't it be nicer to have like, get_ui_option(get_checkbox("Pause", is_paused))?